src/audio/codec_adapter/codec_adapter.c

```diff
@@ -475,8 +475,6 @@ static int codec_adapter_set_params(struct comp_dev *dev, struct sof_ipc_ctrl_da
 					 ret);
 			else
 				comp_dbg(dev, "codec_adapter_set_params() apply of runtime config done.");
-		} else {
-			mod->priv.r_cfg.avail = true;
```

Comment:
How this change is correlated with introducing new 3rd Party Module API? Since there is quite a lot of commits in this PR I would recommend to split clean-up non related commits to separate PR.